OnPageCode_timeout.js

```diff
@@ -1,288 +1,457 @@
+      {
+        key: 'dns',
+        name: 'DNS Lookup',
+        timeEndKey: 'domainLookupEnd',
+        timeStartKey: 'domainLookupStart'
+      },
+      {
+        key: 'tcp',
+        name: 'TCP Connection',
+        timeEndKey: 'connectEnd',
+        timeStartKey: 'connectStart'
+      },
+      {
+        key: 'que_st',
+        name: 'Queueing and stalled',
+        timeEndKey: 'requestStart',
+        timeStartKey: 'startTime'
+      },
+      {
+        key: 'rs_wfs',
+        name: 'Request Sent and Waiting For Server',
+        timeEndKey: 'responseStart',
+        timeStartKey: 'requestStart'
+      },
+      {
+        key: 'cd',
+        name: 'Content Download',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'responseStart'
+      },
+      {
+        key: 'dur',
+        name: 'Duration',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'startTime'
+      }
+    ],
+
+    // bidders to monitor, bidderName => domain to monitor
+    // 'bidders': [
+    //   {
+    //     key: 'pm',
+    //     name: 'PubMatic',
+    //     bidderCode: 'pubmatic',
+    //     searchName: 'hbopenbid.pubmatic.com',
+    //     // reqMethod: 'POST'
+    //   },
+    //   {
+    //     // NOTE: This Object will replace dyanmically depends on dynamic change in translator endpoint config
+    //     key: 'pm',
+    //     name: 'PubMatic',
+    //     bidderCode: 'pubmatic',
+    //     searchName: 'openbidtest-ams.pubmatic.com',
+    //     // reqMethod: 'GET'
+    //   },
+    //   {
+    //     key: 'an',
+    //     name: 'AppNexus',
+    //     bidderCode: 'appnexus',
+    //     searchName: 'ib.adnxs.com'
+    //   },
+    //   {
+    //     key: 'tl',
+    //     name: 'TripleLift',
+    //     bidderCode: 'triplelift',
+    //     searchName: 'tlx.3lift.com'
+    //   },
+    //   {
+    //     key: 'rc',
+    //     name: 'Rubicon',
+    //     bidderCode: 'rubicon',
+    //     searchName: 'fastlane.rubiconproject.com'
+    //   }
+    // ],
+
+    'populateBidderDataForAuction': (bidderObj, requestObj) => {
+      PM_Network_POC.order++;
+      const bidder = PM_Network_POC.bidders.find(bidder => bidder.bidderCode === bidderObj.bidderCode);
+
+      if (!bidder) {
+        PM_Network_POC.bidders.push({
+          bidderCode: bidderObj.bidderCode,
+          searchName: requestObj.url,
+          atLeastOneBidResUsedInAuction: false,
+          bids: [
+            {
+              order: PM_Network_POC.order,
+              reqPayloadCharCount: requestObj.data.length,
+              reqMethod: requestObj.method
+            }
+          ]
+        });
+      } else {
+        bidder.bids.push({
+          order: PM_Network_POC.order,
+          reqPayloadCharCount: requestObj.data.length,
+          reqMethod: requestObj.method
+        });
+      }
     },
-    {
-      key: "cd",
-      name: "Content Download",
-      timeEndKey: "responseEnd",
-      timeStartKey: "responseStart"
+
+    'bidders': [],
```

Comment:
In terms of dynamically adding bidders in this script to track nw data for them per auction, @pm-nitin-shirsat brought up a good point.. as of right now all bidders in an auction will be included in this script, but some auctions may have a large amount of bidders..

Should we whitelist bidders that we would want to dynamically include in this script?  or would we want to just include all participating bidders each time in an auction?

Reply:
@jlquaccia - As I discussed this point with Deepak, he suggested, we can keep dynamic bidders as we are using one or two publishers for our testing and which can have max 10 bidders each. 
But I think, if we need analysis further we may need more publishers which will increase the space in ELK. so it's better to whitelist the bidders. 
your thoughts @pm-harshad-mane ??